### ISSUE TITLE
Is the sided-ness distinction of Mono tracks useful?

### ISSUE THREAD

**Issue by Paul-Licameli** [audacity/audacity#4755]
### Your idea

For all tracks, there is a pan slider (as for any track) which can affect rendering and playback, even if it's mono (try it and listen).

Below the slider you see "Stereo, 44100 Hz" (or whatever rate) for a stereo track.

For a mono track (either made new or by splitting stereo to mono), you see "Mono, 44100 Hz" except when hard panned left or right, in which case it changes to "Left" or "Right", but only at extreme slider settings.  But what's the use?  Why not just "Mono" always?

Now do this:
- Make a Mono track of noise.
- Hard-pan left.
- Copy.
- Deselect all and paste.
- Solo each track, vary its pan slider, and listen.

Observe:  the original plays back panned wherever you like, changing "Mono" to "Left" or "Right" at extremes.

But the copy always says "Left" and varies the volume of the left channel but always plays (or renders) silence in the right.

Besides the failure of copy/paste to make a true copy of all track state -- is there any need for this one-sided kind of track?

### Problem to be solved

Remove some of the legacy rubbish I need to worry about in refactoring.  But visible behavior must change a little bit.

### Prior art

History:  Used to be, there were choices in the track drop-down menu to set this three-state left/mono/right flag in a mono track.

028ed19d908d80fe70bf030089ca8210e75f8af4 way back in 2017 removed those menu items.  "Sync with DarkAudacity" by James.  And yet not all was cleaned up -- I document some remaining weirdness.

I don't recall the rationale either for the original or the change made by James.

**Comment by Paul-Licameli** [comment]
@LWinterberg @Tantacrul ?

**Comment by Paul-Licameli** [comment]
In other words, a mono track depends as if duplicated in two channels, or as if paired with a silent channel, and pans appropriately.

But who really needs that?  The effect can be got by other means (joining two tracks) if you really want it.

**Comment by Tantacrul** [comment]
I'm not sure technically how this should be appropriately implemented although perhaps I can make a few points that might help? Perhaps not?

- Obviously being able to pan Mono tracks is essential (typical to record two guitar or vocal tracks in mono and pan them). I know you are not suggesting we ditch this but thought I'd mention anyway in case anyone misunderstands.
- I agree that to the user, it should just be called a 'mono' track and the panning should not change that. I can't see a value in 'Left' and 'Right' descriptions.
- Exactly how a Mono track should be constructed, I leave to you and @saintmatthieu. Ideally, we'd use an approach commonly used by other applications.

**Comment by Tantacrul** [comment]
One point which most likely is irrelevant to this discussion but I thought I'd mention:

In the future (certainly not now), we would love to allow users to be able to move clips interchangeably between tracks. Clips would be stereo / mono but tracks would just be able to contain either.

Example from Logic:
<img width="1182" alt="image" src="https://github.com/audacity/audacity/assets/21022311/d9570fc3-bea1-4f57-a76e-c1098a436590">

**Comment by Paul-Licameli** [comment]
> I'm not sure technically how this should be appropriately implemented although perhaps I can make a few points that might help? Perhaps not?
> 
> * Obviously being able to pan Mono tracks is essential (typical to record two guitar or vocal tracks in mono and pan them). I know you are not suggesting we ditch this but thought I'd mention anyway in case anyone misunderstands.
> * I agree that to the user, it should just be called a 'mono' track and the panning should not change that. I can't see a value in 'Left' and 'Right' descriptions.
> * Exactly how a Mono track should be constructed, I leave to you and @saintmatthieu. Ideally, we'd use an approach commonly used by other applications.

No argument, panning a mono track to change the balance makes sense.

Question though, whether always silencing one or the other channel as you pan is a useful feature; making a distinction among three kinds of mono tracks (those that silence left, or right , or [default] neither).

Now I don’t think that is valuable — it’s just legacy misfeature, like misalignment of clips in a stereo track — but I want confirmation of that intuition before I go on.

**Comment by Paul-Licameli** [comment]
I’m other words, I want to just abandon this edge case possibility as not valuable, simplifying things — but, I must seek agreement first.

**Comment by Paul-Licameli** [comment]
> One point which most likely is irrelevant to this discussion but I thought I'd mention:
> 
> In the future (certainly not now), we would love to allow users to be able to move clips interchangeably between tracks. Clips would be stereo / mono but tracks would just be able to contain either.
> 
> Example from Logic: <img alt="image" width="1182" src="https://user-images.githubusercontent.com/21022311/242406706-d9570fc3-bea1-4f57-a76e-c1098a436590.png">

Indeed not relevant to this issue, but a good idea that @LWattenberg too lately suggested to me in discussions of the copy-paste cleanup.  I think it makes sense, but not for this release.  It implies a lot for how we should restructure the subdivision of screen area.  See how Audacitybhas duplicate  vertical rulers and a sash between channels?  All that would need redesign and surely seems to me like too much for the present release.

**Comment by Tantacrul** [comment]
If you think it's a misfeature, I have no reason to disagree. I can't imagine why rendering silence in another channel would be needed. Probs best to confirm with the devs but in terms of the feature, I agree with you. Can't see a need.

**Comment by Paul-Licameli** [comment]
Just now I noticed a subtle unintended change of behavior, my fault, at commit 3e0b19d19, committed May 24.

Before that: splitting stereo to mono gave you one track panned hard left and the other hard right.

After:  both are panned center.

Does anyone care?

It can be fixed to behave as before, independently of this issue.

**Comment by Tantacrul** [comment]
Seems like the panning thing is a good idea. Probably best for the result to not audibly change.

**Comment by Paul-Licameli** [comment]
> Seems like the panning thing is a good idea. Probably best for the result to not audibly change.

A different question.

See #4756 ; an unintended change of behavior, and this can be restored, without impairing the rest of the refactoring I must do.

**Comment by saintmatthieu** [comment]
+1 for keeping only one kind of mono track. If we do and someone wants the old behaviour on some track, she can still hard left/right and use the gain slider for loudness.
When opening old projects which have such track, we should probably preserve the hard left- right panning. If pan values are easily mapped to gain values, we should consider being nice and apply extra gain, e.g. gain 0 for a legacy left track hard-panned right.

**Comment by petersampsonaudacity** [comment]
> Seems like the panning thing is a good idea. Probably best for the result to not audibly change.

I **_strongly_** agree with @Tantacrul about this.

**Comment by Paul-Licameli** [comment]
> +1 for keeping only one kind of mono track. If we do and someone wants the old behaviour on some track, she can still hard left/right and use the gain slider for loudness.
> When opening old projects which have such track, we should probably preserve the hard left- right panning. If pan values are easily mapped to gain values, we should consider being nice and apply extra gain, e.g. gain 0 for a legacy left track hard-panned right.

I see this issue is closed, and we have merged the PR that removes this distinction.

Just noting:  in case of legacy projects that do have the distinction, it it simply ignored when opening in 3.4.  I made no effort to make the panning hard left or right in such a case and didn't test it.  I presume few will ever care.
